chromadb/utils/embedding_functions/google_embedding_function.py

```diff
@@ -1,110 +1,341 @@
-import logging
-
-import httpx
-
-from chromadb.api.types import Documents, EmbeddingFunction, Embeddings
-
-logger = logging.getLogger(__name__)
+from chromadb.utils.embedding_functions.embedding_function import (
+    EmbeddingFunction,
+    Space,
+)
+from chromadb.api.types import Embeddings, Documents
+from typing import List, Dict, Any, cast, Optional
+import os
+import numpy as np
+import numpy.typing as npt
 
 
 class GooglePalmEmbeddingFunction(EmbeddingFunction[Documents]):
     """To use this EmbeddingFunction, you must have the google.generativeai Python package installed and have a PaLM API key."""
 
-    def __init__(self, api_key: str, model_name: str = "models/embedding-gecko-001"):
-        if not api_key:
-            raise ValueError("Please provide a PaLM API key.")
-
-        if not model_name:
-            raise ValueError("Please provide the model name.")
+    def __init__(
+        self,
+        api_key_env_var: str = "GOOGLE_PALM_API_KEY",
+        model_name: str = "models/embedding-gecko-001",
+        api_key: Optional[str] = None,
```

Comment:
we should keep the same parameter ordering whenever possible to retain backwards compat if users are using positional params